Fix SMF state machine creation example

kamnxt opened the pull request:
The example further down works, but the first examples have invalid syntax for initializing the state array.

mbolivar-nordic commented:
@kamnxt can you please fix the compliance failures in the checks results? We cannot merge otherwise.

kamnxt committed:
doc: smf: Fix SMF state machine creation example

The example further down works, but the first examples
have invalid syntax for initializing the state array.

Signed-off-by: Kamil Krzyżanowski <kamnxt@kamnxt.com>

kamnxt commented:
@mbolivar-nordic should be fixed now!

kamnxt commented:
:facepalm: oops, done.
